drake/systems/plants/collision/ResultCollector.h

```diff
@@ -0,0 +1,31 @@
+#ifndef DRAKE_SYSTEMS_PLANTS_COLLISION_RESULTCOLLECTOR_H_
+#define DRAKE_SYSTEMS_PLANTS_COLLISION_RESULTCOLLECTOR_H_
+
+#include "drake/systems/plants/collision/DrakeCollision.h"
+
+namespace DrakeCollision {
+class ResultCollector {
+ public:
+  virtual ~ResultCollector() {}
+
+  virtual void addPointPairResult(const PointPair& result);
+
+  inline void addSingleResult(const ElementId idA, const ElementId idB,
+                              const Eigen::Vector3d& ptA,
+                              const Eigen::Vector3d& ptB,
+                              const Eigen::Vector3d& normal,
+                              const double distance) {
+    addPointPairResult(PointPair(idA, idB, ptA, ptB, normal, distance));
+  }
```

Comment:
I would default the other way -- take pity on the people using our APIs and try not to do anything odd there. Programmers can understand these things and adapt easily!